packages/common/container-definitions/src/audience.ts

```diff
@@ -67,4 +73,25 @@ export interface IAudience extends EventEmitter {
 	 * does not technically have a clientId tied to an active connection to the service.
 	 */
 	getMember(clientId: string): IClient | undefined;
+
+	/**
+	 * Returns this client's clientId, if it exists. undefined if this client never connected to ordering service.
+	 * Whenever this property changes, "clientIdChanged" event is fired on this object.
+	 * Wheeever clients loses connection and reconnects, it will raise "connected" event at various API surfaces.
+	 *
+	 * It's guranteed that such events and change of current clientId happens at the same time (syncronously, one after another).
+	 * That said, at the moment this is experimental API. It depends on some experimental settings that might change in the future.
+	 * The claim about clientId being changed simulteniously with raising "connected" event and being up-to-date only valid when
+	 * this API is consumed at loader / hosing layer. Container Runtime layer will get these changes delayed, as loader layer changes
+	 * propagate slowly through the system, and thus newer runtimes will continue to observe old (non-synchronized) behavior when paired
+	 * with old loader
+	 *
+	 * While it's marked as experimental, this promise could be broken, and consumers  could experience current clientId being changed
+	 * (and "clientIdChanged" event fired) while (only applicable for "read" kind of connections)
+	 * 1. Such clientId is not present in Audience
+	 * 2. Client is not fully caught up
+	 *
+	 * @experimental
```

Comment:
Thanks for the detailed docs!

Github didn't play nice with the diff, the suggestion fixes several typos and rewords things a bit.

I don't think we have a great way to say that a particular property in an interface is experimental, but I tried to update to made it clearer and use standard TSDoc constructs.

Also see my previous comment which might result in a few more adjustments here.

```suggestion
	 * (Experimental) Returns this client's clientId, if it exists. undefined if this client has not connected to the ordering service.
	 *
	 * @remarks
	 * This API is experimental.
	 *
	 * Whenever this property changes, the "clientIdChanged" event is fired on this object.
	 * When the clients loses connection and reconnects, it's guaranteed that the "connected" event that fires at various layers, this event, and the change of current clientId, all happen at the same time (synchronously, one after another).
	 * That said, at the moment this is an experimental API. It depends on some experimental settings that might change in the future.
	 * The claim about clientId being changed simultaneously with raising "connected" event and being up-to-date is only valid when
	 * this API is consumed at the loader / hosing layer. The Container Runtime layer will get these changes delayed, as loader layer changes
	 * propagate slowly through the system, and thus newer runtimes will continue to observe old (non-synchronized) behavior when paired
	 * with older loader code.
	 *
	 * While it's marked as experimental, this promise could be broken, and consumers could experience current clientId being changed
	 * (and "clientIdChanged" event fired) while (only applicable for "read" kind of connections)
	 * 1. Such clientId is not present in Audience
	 * 2. Client is not fully caught up
```